### PR TITLE
FIX: ctf data can sometimes have no date time.

### PR DESCRIPTION
**Pull request by kingjr**
I encountered a bug when reading a CTF dataset with an empty string in the `res4['data_date']` and  `res4['data_time']` fields.

This sets the field to January 1st of the year 1001 (earlier make a date parsing bug)

### CODE REVIEW

**Comment by larsoner** [comment]
I would rather set the date to 1) the epoch `(0., 0.)` in FIF form, or 2) None, but (2) probably makes working with things like annotations harder. `(0., 0.)` seems like a better choice than 1001. Or were you trying to avoid possible collisions with anonymized dates...? In that case I could be persuaded :)

**Comment by kingjr** [comment]
new commit: I had to use a higher date, else there is an error at reading time after saving the ctf as a fif. 


Thanks Eric. I did not understand option 1), but I agree that option 2) would be best: we don't have a date, so we shouldn't try to force it. However, it will likely require a lot of downstream corrections.

I'll be using this branch, as it unblocks my pipeline, but feel free to take over.

File: mne/io/ctf/info.py [hunk 1]
```diff
@@ -73,6 +73,11 @@ def _pick_isotrak_and_hpi_coils(res4, coils, t):
 
 def _convert_time(date_str, time_str):
     """Convert date and time strings to float time."""
+    if date_str == time_str == '':
+        date_str = '01/01/2000'
```
Comment (larsoner) [inline mne/io/ctf/info.py:77]:
```suggestion
        date_str = '01/01/1970'
```
This is the epoch and should translate to a `meas_date` of `(0., 0.)`. I'll push a commit to test this code path, too

**Comment by larsoner** [comment]
1970/01/01 okay for you @kingjr ?

@agramfort @hoechenberger @drammock feel free to merge if happy

**Comment by kingjr** [comment]
LGTM, thanks Eric!

File: mne/io/ctf/info.py [hunk 2]
```diff
@@ -73,6 +73,11 @@ def _pick_isotrak_and_hpi_coils(res4, coils, t):
 
 def _convert_time(date_str, time_str):
     """Convert date and time strings to float time."""
+    if date_str == time_str == '':
+        date_str = '01/01/1970'
+        time_str = '00:00:00'
+        logger.info('No date or time found, setting to the epoch (1970/01/01)')
```
Comment (drammock) [inline mne/io/ctf/info.py:79]:
```suggestion
        logger.info('No date or time found, setting to the start of the '
                    'POSIX epoch (1970/01/01 midnight)')
```

**Review by drammock — APPROVED** [review]
LGTM, just one tiny suggestion to make the `logger.info` message clearer.

**Review by agramfort — APPROVED** [review]

**Pull request merged**